packages/parser/src/components/parser/statement/expression/callableExpression.ts

```diff
@@ -0,0 +1,34 @@
+import Expression from ".";
+
+import { TokenTypes } from "../../../../constants/bhaiLangSpec";
+import { NodeType } from "../../../../constants/constants";
+import { ASTNode } from "../../types/nodeTypes";
+
+export default class CallableExpression extends Expression {
+	getExpression(): ASTNode {
+		const name = this._tokenExecutor.eatTokenAndForwardLookahead(TokenTypes.CALLABLE_TYPE).value;
+		this._tokenExecutor.eatTokenAndForwardLookahead(TokenTypes.OPEN_PARENTHESIS_TYPE);
+
+		// read arguments
+		let args: any[] = [];
+		if(this._tokenExecutor.getLookahead()?.type !== TokenTypes.CLOSED_PARENTHESIS_TYPE) {
+			do {
+				args.push(this._getArgs());
+			} while (
+				this._tokenExecutor.getLookahead()?.type === TokenTypes.COMMA_TYPE && 
+				this._tokenExecutor.getLookahead()?.type !== TokenTypes.CLOSED_PARENTHESIS_TYPE&&
```

Comment:
If we have already checked for COMMA_TYPE then we do not need to check for closed PARENTHESIS. First condition is ensuring that it is a COMMA.

Reply:
updated code